src/data/list/rdrop.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2022 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import data.list.basic
+import data.list.infix
+
+/-!
+
+# Dropping or taking from lists on the right
+
+Taking or removing element from the tail end of a list
+
+## Main defintions
+
+- `rdrop n`: drop `n : ℕ` elements from the tail
+- `rtake n`: take `n : ℕ` elements from the tail
+- `drop_while p`: remove all the elements that satisfy a decidable `p : α → Prop` from the tail of
+  a list until hitting the first non-satisfying element
+- `take_while p`: take all the elements that satisfy a decidable `p : α → Prop` from the tail of
+  a list until hitting the first non-satisfying element
+
+## Implementation detail
+
+These four methods operate by performing the regular "from-left" operation on `list.reverse`,
+followed by another `list.reverse`, so they are not the most performant.
+
+-/
+
+variables {α : Type*} (p : α → Prop) [decidable_pred p] (l : list α) (n : ℕ)
+
+namespace list
+
+/-- Drop `n` elements from the tail end of a list. Implemented naively via `list.reverse` -/
+def rdrop : list α := reverse (l.reverse.drop n)
```

Comment:
I guess this definition is nice for proving things. But wouldn't it be better to invest a little bit in get a performant definition? Proving that it is equivalent to this shouldn't be too hard.

Reply:
What would be more performant? Given some proof that the `l.length = L`, construct a list that is from the first `L - n` elements?

Reply:
Hmm, I don't know anything about performance. But the comments in this file suggested to me that you knew about more performant (but maybe uglier to implement) versions...

If not: lgtm!

bors d+

Reply:
Sure, I can express it as the `list.take l (l.length - n)` version.

Reply:
I'll leave it up to you.